src/planner/binder/tableref/plan_joinref.cpp

```diff
@@ -324,7 +324,10 @@ unique_ptr<LogicalOperator> Binder::CreatePlan(BoundJoinRef &ref) {
 	// now create the join operator from the join condition
 	auto result = LogicalComparisonJoin::CreateJoin(context, ref.type, ref.ref_type, std::move(left), std::move(right),
 	                                                std::move(ref.condition));
-
+	if (ref.type == JoinType::MARK) {
```

Comment:
Can this be moved down and access `join` instead? `result->Cast<LogicalJoin>()` will fail if the result is a `LogicalFilter` which seems like it can happen according to the code below